src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -2181,6 +2184,30 @@ public void selectAll() {
             // this is done to restore the anchor and lead
             SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
 
+            selModel.setValueIsAdjusting(false);
+        } else if (rowCount > 0 && columnCount == 0) {
+            selModel = selectionModel;
+            selModel.setValueIsAdjusting(true);
+            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
+            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
+
+            setRowSelectionInterval(0, getRowCount()-1);
+
+            // this is done to restore the anchor and lead
+            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+
+            selModel.setValueIsAdjusting(false);
```

Comment:
Recently added code is a subset of the code block in `if block`. Duplicate code can be refactored with helper methods.

Reply:
Duplication removed